services/core/shared/core/delegates.js

```diff
@@ -122,20 +122,24 @@ const getDelegates = async params => {
 	const sortComparator = (sortParam) => {
 		const sortProp = sortParam.split(':')[0];
 		const sortOrder = sortParam.split(':')[1];
-
-		const comparator = (a, b) => {
-			try {
-				if (Number.isNaN(Number(a[sortProp]))) throw new Error('Not a number, try string sorting');
-				return (sortOrder === 'asc')
-					? a[sortProp] - b[sortProp]
-					: b[sortProp] - a[sortProp];
-			} catch (_) {
-				return (sortOrder === 'asc')
-					? a[sortProp].localeCompare(b[sortProp])
-					: b[sortProp].localeCompare(a[sortProp]);
-			}
-		};
-		return comparator;
+		try {
+			const comparator = (a, b) => {
+				try {
+					if (Number.isNaN(Number(a[sortProp]))) throw new Error('Not a number, try string sorting');
+					return (sortOrder === 'asc')
+						? a[sortProp] - b[sortProp]
+						: b[sortProp] - a[sortProp];
+				} catch (_) {
+					return (sortOrder === 'asc')
+						? a[sortProp].localeCompare(b[sortProp])
+						: b[sortProp].localeCompare(a[sortProp]);
+				}
+			};
+			return comparator;
+		} catch (err) {
+			logger.error(err);
+			throw err;
+		}
```

Comment:
Why is that logic updated?

Reply:
Sorry, it seems like I wasn't able to communicate it clearly. The idea was to wrap the following piece of code and debug the error stack and determine the conditions of when we encounter the error `Cannot read property 'localeCompare' of undefined`.

We were not supposed to check in this piece of code.
```
					return (sortOrder === 'asc')
						? a[sortProp].localeCompare(b[sortProp])
						: b[sortProp].localeCompare(a[sortProp]);
```

Reply:
Reverting it back!